packages/engine/src/scene/systems/ShadowSystem.ts

```diff
@@ -142,62 +144,85 @@ export default async function ShadowSystem(world: World) {
   let dropShadows = new InstancedMesh(shadowGeometry, shadowMaterial, 0)
   dropShadows.matrixAutoUpdate = false
 
-  const dropShadowReactor = startQueryReactor([DropShadowComponent, GroupComponent], function modifyShadowCount() {
-    world.scene.remove(dropShadows)
-    dropShadows = new InstancedMesh(shadowGeometry, shadowMaterial, shadowComponentQuery().length)
-    dropShadows.matrixAutoUpdate = false
-    dropShadows.layers.disable(ObjectLayers.Camera)
-    world.scene.add(dropShadows)
+  const shadowComponentQuery = defineQuery([DropShadowComponent, GroupComponent])
+
+  let sceneObjects = Array.from(Engine.instance.currentWorld.objectLayerList[ObjectLayers.Camera] || [])
+
+  const dropShadowReactor = startQueryReactor([DropShadowComponent, GroupComponent], function (props) {
+    const dropShadowComponent = useComponent(props.root.entity, DropShadowComponent)
+    const groupComponent = useComponent(props.root.entity, GroupComponent)
+    const useShadows = useShadowsEnabled()
+
+    useEffect(() => {
+      const getShadows = getShadowsEnabled()
```

Comment:
No need to have an extra call here, we can use our previously define useShadows instead